tests/examples/examples_ft.cpp

```diff
@@ -0,0 +1,218 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <iostream> // TODO remove
+#include <string>
+#include <regex>
+#include <fstream>
+#include <vector>
+#include <k4a/k4a.h>
+#include <utcommon.h>
+#include <gtest/gtest.h>
+
+#ifdef _WIN32
+#define PATH_TO_BIN(binary) "bin\\" binary ".exe"
+#define MKDIR(path) "if not exist " + path + " mkdir " + path
+#define RMDIR(path) "rmdir /S /Q " + path
+#define POPEN _popen
+#define PCLOSE _pclose
+#define SETENV(env, value) _putenv_s(env, value)
+#else
+#define PATH_TO_BIN(binary) "./bin/" binary
+#define MKDIR(path) "mkdir -p " + path
+#define RMDIR(path) "rm -rf " + path
+#define POPEN popen
+#define PCLOSE pclose
+#define SETENV(env, value) setenv(env, value, 1)
+#endif
+
+const static std::string TEST_TEMP_DIR = "examples_test_temp";
+
+// run the specified executable and record the output to output_path
+// if output_path is empty, just output to stdout
+static void run_and_record_executable(const std::string &shell_command_path, const std::string &output_path)
+{
+    // TODO fflush needed here?
+    std::string formatted_command = shell_command_path;
+    if (!output_path.empty())
+    {
+        formatted_command += " 2>&1 > " + output_path;
+    }
+    std::cout << formatted_command << std::endl;
+    FILE *process_stream = POPEN(formatted_command.c_str(), "r");
```

Comment:
Can you confirm the exit code of the application?

Reply:
pclose() returns the exit status, which is now returned from the function and asserted to EXIT_SUCCESS.